sql/core/benchmarks/CollationBenchmark-jdk21-results.txt

```diff
@@ -2,26 +2,53 @@ OpenJDK 64-Bit Server VM 21.0.2+13-LTS on Linux 6.5.0-1017-azure
+UTF8_BINARY                                          4335           4338           4          0.0       43354.3       1.6X
+UNICODE_CI                                          44575          44582           9          0.0      445751.5       0.2X
 
 OpenJDK 64-Bit Server VM 21.0.2+13-LTS on Linux 6.5.0-1017-azure
 AMD EPYC 7763 64-Core Processor
 collation unit benchmarks - compareFunction:  Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
 ---------------------------------------------------------------------------------------------------------------------------
-UTF8_BINARY_LCASE                                     6522           6526           4          0.0       65223.9       1.0X
-UNICODE                                              45792          45797           7          0.0      457922.3       0.1X
-UTF8_BINARY                                           7092           7112          29          0.0       70921.7       0.9X
-UNICODE_CI                                           47548          47564          22          0.0      475476.7       0.1X
+UTF8_BINARY_LCASE                                     7184           7201          25          0.0       71837.4       1.0X
+UNICODE                                              46620          46651          43          0.0      466199.7       0.2X
+UTF8_BINARY                                           7110           7115           7          0.0       71104.4       1.0X
+UNICODE_CI                                           47808          47835          37          0.0      478084.9       0.2X
 
 OpenJDK 64-Bit Server VM 21.0.2+13-LTS on Linux 6.5.0-1017-azure
 AMD EPYC 7763 64-Core Processor
 collation unit benchmarks - hashFunction:  Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
 ------------------------------------------------------------------------------------------------------------------------
-UTF8_BINARY_LCASE                                 11716          11716           1          0.0      117157.9       1.0X
-UNICODE                                          180133         180137           5          0.0     1801332.1       0.1X
-UTF8_BINARY                                       10476          10477           1          0.0      104757.4       1.1X
-UNICODE_CI                                       148171         148190          28          0.0     1481705.6       0.1X
+UTF8_BINARY_LCASE                                 11380          11390          15          0.0      113795.1       1.0X
+UNICODE                                          176471         176477           9          0.0     1764709.9       0.1X
+UTF8_BINARY                                       10858          10859           1          0.0      108580.4       1.0X
+UNICODE_CI                                       153060         153161         143          0.0     1530598.1       0.1X
```

Comment:
I see that the benchmark runs on previous functions too, but we shouldn't make any unnecessary modifications in this PR (that are not related to the changes brought by this PR)

so I would advise keep the previous results as they were, and only append yours below